core/src/avm2/object/namespace_object.rs

```diff
@@ -65,7 +75,15 @@ impl<'gc> NamespaceObject<'gc> {
 
         let this: Object<'gc> = NamespaceObject(GcCell::new(
             activation.context.gc_context,
-            NamespaceObjectData { base, namespace },
+            NamespaceObjectData {
+                base,
+                namespace,
+                prefix: if namespace.as_uri_opt().is_some() {
```

Comment:
Any reason this is setting the prefix based on if the URI is `Some` and not like it is done in the constructor (checking if URI is empty)?

Reply:
Because I wrote that first and forgot to come back and update it :D